src/test/java/games/strategy/net/MessengerTestUtils.java

```diff
@@ -0,0 +1,25 @@
+package games.strategy.net;
+
+import javax.annotation.Nullable;
+
+/**
+ * A collection of methods useful for writing tests that use instances of {@link IMessenger}.
+ */
+public final class MessengerTestUtils {
+  private MessengerTestUtils() {}
+
+  /**
+   * Shuts down the specified messenger unconditionally. Any exception will be written to stdout.
+   *
+   * @param messenger The messenger to shut down; if {@code null}, this method does nothing.
+   */
+  public static void shutDownQuietly(final @Nullable IMessenger messenger) {
+    if (messenger != null) {
+      try {
+        messenger.shutDown();
```

Comment:
The other question is: Do we really want to suppress any exceptions when shutting down?
If this throws an exception this indicates that tests weren't properly written, especially because we can use mocks.

Reply:
I would say, no we don't want to suppress exceptions when shutting down (or anyplace, for that matter).  Before submitting this PR, I was tempted to just remove the try-catch blocks.  However, after reading some of the comments in the affected tests, I got the impression that they are Flaky Tests, and suppressing the exceptions at shutdown was a hack to workaround some race condition between the various `IMessenger`s disconnecting from each other.

We should probably investigate this further, but I believe it's out of scope for this PR.  I think your suggestion of fully mocking the behavior in the unit test cases is probably the right thing to do.